contributed_definitions/nyaml/NXpositioner_spm.yaml

```diff
@@ -1,242 +1,163 @@
+
 type: group
-NXpositioner_sts(NXobject):
-  name:
-    doc: |
-      symbolic or mnemonic name (one word)
-  description:
-    doc: |
-      description of positioner
-  value(NX_NUMBER):
-    unit: NX_ANY
-    doc: |
-      best known value of positioner - need [n] as may be scanned
-    dimensions:
-      rank: 1
-      dim: [[1, n]]
-  raw_value(NX_NUMBER):
-    unit: NX_ANY
-    doc: |
-      raw value of positioner - need [n] as may be scanned
-    dimensions:
-      rank: 1
-      dim: [[1, n]]
-  target_value(NX_NUMBER):
-    unit: NX_ANY
-    doc: |
-      targeted (commanded) value of positioner - need [n] as may be scanned
-    dimensions:
-      rank: 1
-      dim: [[1, n]]
-  tolerance(NX_NUMBER):
-    unit: NX_ANY
-    doc: |
-      maximum allowable difference between target_value and value
-    dimensions:
-      rank: 1
-      dim: [[1, n]]
-  soft_limit_min(NX_NUMBER):
-    unit: NX_ANY
-    doc: |
-      minimum allowed limit to set value
-  soft_limit_max(NX_NUMBER):
-    unit: NX_ANY
-    doc: |
-      maximum allowed limit to set value
-  velocity(NX_NUMBER):
-    unit: NX_ANY
-    doc: |
-      velocity of the positioner (distance moved per unit time)
-  acceleration_time(NX_NUMBER):
-    unit: NX_ANY
-    doc: |
-      time to ramp the velocity up to full speed
-  controller_record:
-    doc: |
-      Hardware device record, e.g. EPICS process variable, taco/tango ...
-  \@default:
-    doc: |
-      .. index:: plotting
-      
-      Declares which child group contains a path leading
-      to a :ref:`NXdata` group.
-      
-      It is recommended (as of NIAC2014) to use this attribute
-      to help define the path to the default dataset to be plotted.
-      See https://www.nexusformat.org/2014_How_to_find_default_data.html
-      for a summary of the discussion.
-  depends_on(NX_CHAR):
-    doc: |
-      NeXus positions components by applying a set of translations and rotations
-      to apply to the component starting from 0, 0, 0. The order of these operations
-      is critical and forms what NeXus calls a dependency chain. The depends_on
-      field defines the path to the top most operation of the dependency chain or the
-      string "." if located in the origin. Usually these operations are stored in a
-      NXtransformations group. But NeXus allows them to be stored anywhere.
-      
-      .. todo::
-         Add a definition for the reference point of a positioner.
-  (NXtransformations):
-    doc: |
-      This is the group recommended for holding the chain of translation
-      and rotation operations necessary to position the component within
-      the instrument. The dependency chain may however traverse similar groups in
-      other component groups.
-  position(NXtransformations):
-    doc: |
-      To clarify the frame laboratory frame. The scanning area in x, y, and z position in the
-      frame.
-  z_contronller(NX_NUMBER):
+NXpositioner_spm(NXpositioner):
```

Comment:
This extends a base class, should be kept in mind that this is not yet officially endorsed by NIAC.